email-common/src/main/java/io/commercetools/sunrise/email/EmailSender.java

```diff
@@ -44,15 +43,19 @@
      * }</pre>
      * Exceptions that occur while sending the email are instances of {@link EmailSenderException} that are
      * wrapped in an unchecked {@link CompletionException} thrown by {@link CompletableFuture#join()}.
-     * <p>Implementations of this method need to apply timeouts that may be configured when creating the {@link EmailSender}
+     * <p>
+     * Implementations of this method need to apply timeouts that may be configured when creating the {@link EmailSender}
      * instance. The timeouts avoid denial of service by too many connections waiting for stalled I/O.</p>
      *
-     * @param messageAuthor a consumer that fills the empty message created by the email sender.
-     *                      Note that {@link MimeMessage} instances are not immutable. Messages passed to the consumer
-     *                      must only be used by that consumer instance and must not be passed elsewhere.
+     * @param messageEditor the email sender passes an empty message to the message editor that the editor shall fill.
+     *                      Note that {@link MimeMessage} instances are not immutable. Messages passed to the editor
+     *                      must only be used by that editor instance and must not be passed elsewhere.
      * @return A completion stage for sending the message asynchronously.
+     * @throws EmailSenderException if there was an error while creating or filling the message. Note that in contrast
```

Comment:
I would create 2 different exceptions to avoid confusion, one for the email creation and another for the email sending. Email creation exception will be directly thrown and the Email sending exception will be thrown wrapped in the `CompletionStage`. That way we can have two different `@throws` documentation and be clearer about it.